src/query/interpret/awesome_memgraph_functions.cpp

```diff
@@ -1174,6 +1177,46 @@ TypedValue Duration(const TypedValue *args, int64_t nargs, const FunctionContext
   MapNumericParameters<Number>(parameter_mappings, args[0].ValueMap());
   return TypedValue(utils::Duration(duration_parameters), ctx.memory);
 }
+
+std::function<TypedValue(const TypedValue *, const int64_t, const FunctionContext &)> UserFunction(
+    const procedure::ModulePtr &module_ptr, const mgp_func &func, const std::string &fully_qualified_name) {
+  /// Module pointer should be propagated because of the lock aquired. It prevents reloading module while function is
+  /// executing.
+
+  // NOLINTNEXTLINE(clang-diagnostic-unused-lambda-capture)
+  return [func, fully_qualified_name, &module_ptr](const TypedValue *args, int64_t nargs,
```

Comment:
```suggestion
std::function<TypedValue(const TypedValue *, const int64_t, const FunctionContext &)> UserFunction(
    procedure::ModulePtr &&module_ptr, const mgp_func &func, const std::string &fully_qualified_name) {
  /// Module pointer should be propagated because of the lock aquired. It prevents reloading module while function is
  /// executing.

  // NOLINTNEXTLINE(clang-diagnostic-unused-lambda-capture)
  return [func, fully_qualified_name, module_ptr = std::move(module_ptr)](const TypedValue *args, int64_t nargs,
```

You have to move the `module_ptr`, because a raw reference won't prevent destruction.